packages/language-service/src/typescript_host.ts

```diff
@@ -289,37 +287,37 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
    * Returns true if modules are up-to-date, false otherwise.
    * This should only be called by getAnalyzedModules().
    */
-  private upToDate() {
-    const program = this.program;
-    if (this.lastProgram === program) {
+  private upToDate(): boolean {
+    const {lastProgram, program} = this;
+    if (lastProgram === program) {
       return true;
     }
+    this.lastProgram = program;
 
     // Invalidate file that have changed in the static symbol resolver
     const seen = new Set<string>();
-    let hasChanges = false;
     for (const sourceFile of program.getSourceFiles()) {
       const fileName = sourceFile.fileName;
       seen.add(fileName);
       const version = this.tsLsHost.getScriptVersion(fileName);
       const lastVersion = this.fileVersions.get(fileName);
-      if (version !== lastVersion) {
-        hasChanges = true;
-        this.fileVersions.set(fileName, version);
-        this.staticSymbolResolver.invalidateFile(fileName);
+      this.fileVersions.set(fileName, version);
+      // Should not invalidate file on the first encounter
```

Comment:
nit: otherwise the comment makes it seem like all of `lastVersion !== undefined && version !== lastVersion` is needed to see if the file is encountered for the first time

```suggestion
      // Should not invalidate file on the first encounter or if file hasn't changed
```